### ISSUE TITLE
Setting horizontal or vertical alignment, but not both, affects how the element is measured in the other direction.

### ISSUE THREAD

**Issue by zlatanov** [QuestPDF/QuestPDF#401]
**Describe the bug**
If we only set either horizontal or vertical alignment of a component, because both of them are implemented in a single `ContainerElement` and we do not have a default `Stretch` or `Unset` value, then the component is rendered as if we've explicitly set both.

**To Reproduce**
```c#
using QuestPDF.Fluent;
using QuestPDF.Helpers;
using QuestPDF.Infrastructure;

var document = Document.Create(container =>
{
    container.Page(page =>
    {
        page.Size(PageSizes.A4);
        page.Content().Column(column =>
        {
            column.Item().Height(300).Component(new Component());
        });
    });
});

class Component : IComponent
{
    public void Compose(IContainer container)
    {
        container.AlignCenter()
                 .Background("#FF0000")
                 .Text("TEST");
    }
}
```

**Expected behavior**
I would expect if I aligned a component either horizontally or vertically, the other direction would not be affected.

**Screenshots**
Expected:
![image](https://user-images.githubusercontent.com/2470527/200308817-55652ba7-92b8-4664-937a-c027de16c91e.png)

Actual:
![image](https://user-images.githubusercontent.com/2470527/200308972-3f4589f6-0201-46c3-ba6c-e220d37a458f.png)

**Comment by zlatanov** [comment]
I've pinpointed the issue to the following line:
https://github.com/QuestPDF/QuestPDF/blob/983edd5f7667406e4a12035b2c6585e6024433d3/QuestPDF/Elements/Alignment.cs#L26

Basically, because the alignment element thinks we've set both Horizontal and Vertical directions, then the draw must use the measured child size and should not `stretch` to the available size.

However, I think that, as in my example, is incorrect and the `base.Draw` should only restrict the size in the direction of the alignments explicitly set. This though seems like a breaking change and might cause documents rendered with the current version to render differently.

I am happy to use any workaround, but since the rendering behaviors are internal, I cannot reimplement the `Alignment`.

**Comment by MarcinZiabek** [comment]
That's interesting! I need to think more about it. At this moment, I tend to agree with you regarding the desired behavior. 

This is a breaking change. If we decide to make a proper improvement, it should be published within the next major release in December. Till then, let's think about potential pros and cons of current and new behaviours. And if there are any similar problems anywhere else.

This may be a nice additional to the following PR: https://github.com/QuestPDF/QuestPDF/pull/357 Would you like to take a look and present your thoughts?

**Comment by zlatanov** [comment]
Thanks @MarcinZiabek, I will take a careful look at the PR this weekend!

**Comment by MarcinZiabek** [comment]
I have fixed this issue with the commit https://github.com/QuestPDF/QuestPDF/commit/d2ba21821f8956208167f3d91e0f8890720f12e3. It will be released along with the 2022.12 version.

Thank you for noticing this problem 😁
